reframe/utility/dependencies.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
```

Comment:
We need to document this module and include it somewhere inside the "ReFrame manuals" section.

Would it also make sense to rename this package to `udeps` for "micro-dependencies." I don't know if you had this in mind when you included as `udeps` elsewhere, but I like it as an idea.